### PR TITLE
Issue #6207: Add Xpath Regression Test for RedundantImport

### PR DESCRIPTION
**Pull request by Binita-tech**
Partial for Issue: #6207

Add xpath regression test for RedundantImport

### CODE REVIEW

**Review by Vyom-Yadav — CHANGES_REQUESTED** [review]
Please do relevant changes in all the tests.

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+    @Test
+    public void testOne(List<String> expectedXpathQueries) throws Exception {
+        final File fileToProcess =
+                new File(getPath("SuppressionXpathRegressionRedundantImport1.java"));
+        final DefaultConfiguration moduleConfig =
+                createModuleConfig(RedundantImportCheck.class);
+        final String[] expectedViolation = {
+                "3:104: " + getCheckMessage(RedundantImportCheck.class,
+                        RedundantImportCheck.MSG_KEY, "org.checkstyle.suppressionxpathfilter.redundantimport.SuppressionXpathRegressionRedundantImport1"),
```
Comment (Vyom-Yadav) [inline src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java:48]: `RedundantImportCheck` doesn't have any variable named `MSG_KEY`, it has `MSG_LANG`, `MSG_SAME`, `MSG_DUPLICATE`. A message key points to the complete error message in `.properties` file, example-
```
public static final String MSG_LANG = "import.lang";
```
`import.lang` is pointing to complete message in `.properties` file
https://github.com/checkstyle/checkstyle/blob/b68f16193f4db8b964797ae3a708aac29b296365/src/main/resources/com/puppycrawl/tools/checkstyle/checks/imports/messages.properties#L15

This key-value pair helps us show error messages in multiple languages-
https://github.com/checkstyle/checkstyle/blob/b68f16193f4db8b964797ae3a708aac29b296365/src/main/resources/com/puppycrawl/tools/checkstyle/checks/imports/messages_fr.properties#L15

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        };
+        final List<String> expectedXPathQueries = Collections.singletonList(
+                "/IMPORT/DOT[./IDENT[@text='SuppressionXpathRegressionRedundantImport1']]/DOT[./IDENT[@text='redundantimport']]"
+                        + "/DOT[./IDENT[@text='suppressionxpathfilter']]/DOT[./IDENT[@text='checkstyle']]"
+                        + "/DOT/IDENT[@text='org']");
```
Comment (Vyom-Yadav) [inline src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java:53]: This Xpath query isn't right, it should be `/COMPILATION_UNIT/IMPORT`

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    private final String checkName = RedundantImportCheck.class.getSimpleName();
+
+    @Override
+    protected String getCheckName() {
+        return checkName;
+    }
+
+    @Test
+    public void testOne(List<String> expectedXpathQueries) throws Exception {
```
[the remark this answers is not in the document]
Comment (Binita-tech) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java:41]: `runVerifications(moduleConfig, fileToProcess, expectedViolation,
                expectedXpathQueries);`
I am getting error for expectedXpathQueries when I removing parameter @Vyom-Yadav

[the remark this answers is not in the document]
Comment (Vyom-Yadav) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java:41]: Please share what error you are getting.

[the remark this answers is not in the document]
Comment (Binita-tech) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java:41]: [ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /C:/Users/Dell/Desktop/G_Checkstyle/checkstyle/src/it/java/org/checkstyl
                                                                               le/suppressionxpathfilter/XpathRegressionRedundantImportTest.java:[56,17] cannot 
                                                                                find symbol
  symbol:   variable expectedXpathQueries
  location: class org.checkstyle.suppressionxpathfilter.XpathRegressionRedundant
                                                                               tImportTest
[ERROR] /C:/Users/Dell/Desktop/G_Checkstyle/checkstyle/src/it/java/org/checkstyl
                                                                               le/suppressionxpathfilter/XpathRegressionRedundantImportTest.java:[73,17] cannot 
                                                                                find symbol
  symbol:   variable expectedXpathQueries

[the remark this answers is not in the document]
Comment (Vyom-Yadav) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java:41]: Cannot find symbol simply means there isn't a variable with that name, do you use an IDE?

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        final File fileToProcess =
+                new File(getPath("SuppressionXpathRegressionRedundantImport1.java"));
+        final DefaultConfiguration moduleConfig =
+                createModuleConfig(RedundantImportCheck.class);
+        final String[] expectedViolation = {
+                "3:104: " + getCheckMessage(RedundantImportCheck.class,
```
[the remark this answers is not in the document]
Comment (Binita-tech) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java:47]: Here 3 is row no and 104 is column no right, I checked I am getting 3:104, please tell me where I am going wrong?

[the remark this answers is not in the document]
Comment (Vyom-Yadav) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java:47]: In the file, `SuppressionXpathRegressionRedundantImport1.java` violation will be on line 3 and column 1, if you are getting different output, please share the output with config.xml and your test file, make sure you are using the latest jar.
```
[ERROR] /home/vyom/IdeaProjects/ConfigCheckstyle/src/TestP/Test.java:3:1: Redundant import from the same package - org.checkstyle.suppressionxpathfilter.redundantimport.SuppressionXpathRegressionRedundantImport1. [RedundantImport]
```

[the remark this answers is not in the document]
Comment (Binita-tech) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java:47]: Now, I understood, I had misunderstood your words. I changed row no and column no in all tests.

 `[ERROR] testThree  Time elapsed: 0.001 s  <<< ERROR!
java.lang.IllegalArgumentException: Cannot format given Object as a Number
        at org.checkstyle.suppressionxpathfilter.XpathRegressionRedundantImportTest.testThree(XpathRegressionRedundantImportTest.java:83)`
I am getting this error,  can you tell why I am getting this?



>

[the remark this answers is not in the document]
Comment (Vyom-Yadav) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java:47]: I can but I won't, find it yourself, examine the stack trace and try to figure out what's happening. Also, update the code on the remote branch as well so I can the actual changes.

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java [hunk 5]
```diff
@@ -0,0 +1,94 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2021 the original author or authors.
```
[the remark this answers is not in the document]
Comment (Binita-tech) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java]: Done

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package org.checkstyle.suppressionxpathfilter;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.Test;
```
[the remark this answers is not in the document]
Comment (Binita-tech) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java]: Done

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java (hunk 1 above)
Comment (Binita-tech) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java]: Done

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java (hunk 2 above)
Comment (Binita-tech) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java]: Done

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (Vyom-Yadav) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java]: Resolved

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java (hunk 6 above)
[the remark this answers is not in the document]
Comment (Vyom-Yadav) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java]: Resolved

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (Vyom-Yadav) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java]: Resolved

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java (hunk 1 above)
Comment (Vyom-Yadav) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java]: Resolved

**Comment by romani** [comment]
Please make sure "mvn verify" is passing on your local.

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (Binita-tech) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java]: @Vyom-Yadav, Can you give some hints, I tried but was not able to resolve this error?

[the remark this answers is not in the document]
Comment (Vyom-Yadav) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java]: The error is on line 83 when you invoke the `getCheckMessage()` method, you are only passing one argument whereas `MSG_DUPLICATE` expects 2 with the first one being an integer hence getting the error `Cannot format given Object as a Number at..` as the first argument you pass is `"java.util.Scanner"`. If you open the respected `.properties` file you can see-
https://github.com/checkstyle/checkstyle/blob/c154d616872d6a8bdb41089dca5df8ebbe9984d0/src/main/resources/com/puppycrawl/tools/checkstyle/checks/imports/messages.properties#L12

[the remark this answers is not in the document]
Comment (Binita-tech) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java]: Done

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java (hunk 2 above)
Comment (Vyom-Yadav) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java]: Resolved

File: src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (Vyom-Yadav) [reply src/it/java/org/checkstyle/suppressionxpathfilter/XpathRegressionRedundantImportTest.java]: Resolved

**Review by Vyom-Yadav — APPROVED** [review]
Good job!

**Review by romani — APPROVED** [review]
Impressive cooperation of contributors

**Pull request merged**